pkg/validation/internal/resources.go

```diff
@@ -0,0 +1,76 @@
+package internal
+
+import (
+	goerrors "errors"
+	"fmt"
+
+	"github.com/operator-framework/api/pkg/manifests"
+	operatorsv1alpha1 "github.com/operator-framework/api/pkg/operators/v1alpha1"
+	"github.com/operator-framework/api/pkg/validation/errors"
+	interfaces "github.com/operator-framework/api/pkg/validation/interfaces"
+)
+
+
+// ResourcesValidator validates the resource configuration defined in the CSV
+//
+// This validator will raise an WARNING when the resources request for CPU and/or Memory are not
+// defined for any of the containers found in the CSV
+var ResourcesValidator interfaces.Validator = interfaces.ValidatorFunc(validateResourcesValidator)
+
+func validateResourcesValidator(objs ...interface{}) (results []errors.ManifestResult) {
+	for _, obj := range objs {
+		switch v := obj.(type) {
+		case *manifests.Bundle:
+			results = append(results, validateResources(v))
+		}
+	}
+
+	return results
+}
+
+func validateResources(bundle *manifests.Bundle) errors.ManifestResult {
+	result := errors.ManifestResult{Name: bundle.Name}
+
+	if bundle == nil {
+		result.Add(errors.ErrInvalidBundle("Bundle is nil", nil))
+		return result
+	}
+
+	if bundle.CSV == nil {
+		result.Add(errors.ErrInvalidBundle("Bundle csv is nil", bundle.Name))
+		return result
+	}
+
+	errs, warns := validateResourceRequests(bundle.CSV)
+	for _, err := range errs {
+		result.Add(errors.ErrFailedValidation(err.Error(), bundle.CSV.GetName()))
+	}
+	for _, warn := range warns {
+		result.Add(errors.WarnFailedValidation(warn.Error(), bundle.CSV.GetName()))
+	}
+
+	return result
+}
+
+// validateResourceRequests will return a WARN when the resource request is not set
+func validateResourceRequests(csv *operatorsv1alpha1.ClusterServiceVersion) (errs, warns []error) {
+	if csv.Spec.InstallStrategy.StrategySpec.DeploymentSpecs == nil {
+		errs = append(errs, goerrors.New("unable to find a deployment to install in the CSV"))
+		return errs, warns
+	}
+	deploymentSpec := csv.Spec.InstallStrategy.StrategySpec.DeploymentSpecs
+
+	for _, dSpec := range deploymentSpec {
+		for _, c := range dSpec.Spec.Template.Spec.Containers {
+			if c.Resources.Requests == nil || !(len(c.Resources.Requests.Cpu().String()) != 0 && len(c.Resources.Requests.Memory().String()) != 0) {
+				msg := fmt.Errorf("it is recommended set the resource requests for all containers " +
+					"created and managed via the Operator project. Note that the container %s does not " +
+					"have a resource request for CPU and/or Memory define. Inform the minimal general values required " +
+					"to ensure that the containers will have them on the cluster. " +
+					"More info: https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/", c.Name)
+				warns = append(warns, msg)
```

Comment:
/hold

Memory and CPU requests and limits are associated with Containers, but be aware that the Memory and CPU 
requests and limits of a Pod are the sums of its specific computing types for all the containers in the Pod. 

We need to change it to be the sum

Reply:
/hold cancel 

@fgiloux dis some checks and:

> But if a LimitRange constraint at the Pod level or a Quota is specified all the containers of a pod need to provide resource requests or limits otherwise the Pod gets rejected

So, the validator here shows right we need to warn for all containers.